src/Normalizer.php

```diff
@@ -85,7 +85,7 @@ public static function normalize(
             }
 
             if ($flags & self::REDUCE_DUPLICATE_PATH_SLASHES) {
-                $uri->withPath(preg_replace('#//++#', '/', $uri->getPath()));
+                $uri = $uri->withPath(preg_replace('#//+#', '/', $uri->getPath()));
```

Comment:
The assignment is indeed an error. The regex itself is correct. The unit tests for this are incorrect.

The regex should be reverted to what is once was and the [relevant test case](https://github.com/webignition/uri/blob/master/tests/NormalizerTest.php#L299) needs a minor tweak to reflect the correct behavior.

Reply:
I've reverted the regex and fixed the test.